### PR TITLE
feat: move to vitejs

### PR DESCRIPTION
**Pull request by spaenleh**
In this PR:
- move to **vite** for bundling
  - enjoy _near instant_ dev server startup ⏩ 
  - no more eslint plugin hell 😈 
  - faster builds 🏗️ 
- fix _a lot_ of ts errors in cypress test files 🩹 🚑 
- tweak the cypress workflow to play nicely with vite:
  1. build the app in `test` mode with coverage (had to increase the memory limit ...)
  2. start cypress tests with the app in "preview" mode (the built code is served with an http server)
  - this eliminates some failing tests related to WebSockets mocking. Vite uses WebSockets in its dev-server to push HMR updates, if we mock the WebSockets it breaks the tests... Using a build without a dev server eliminates this issue.

Collateral dammage:
- faster CI: before 45min, now: 30min

To be addressed in up-coming PRs:
- #735

### CODE REVIEW

File: .github/workflows/deploy-prod.yml [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+#     # Reference reusable workflow file. Using the commit SHA is the safest for stability and security
+#     uses: graasp/graasp-deploy/.github/workflows/cdeployment-s3.yml@v1
+#     # Replace input build-folder if needed.
+#     with:
+#       build-folder: 'build'
+#       tag: ${{ github.event.client_payload.tag }}
+#     # Insert required secrets based on repository with the following format: ${{ secrets.SECRET_NAME }}
+#     secrets:
+#       api-host: ${{ secrets.REACT_APP_API_HOST_PROD }}
+#       authentication-host: ${{ secrets.REACT_APP_AUTHENTICATION_HOST_PROD }}
```
Comment (pyphilia) [inline .github/workflows/deploy-prod.yml:74]: delete? what about GRAASP_PLAYER_HOST, etc?

File: .github/workflows/deploy-stage.yml [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+#     name: Graasp Builder
+#     uses: graasp/graasp-deploy/.github/workflows/cdelivery-s3.yml@v1
+#     with:
+#       build-folder: 'build'
+#       tag: ${{ github.event.client_payload.tag }}
+#     secrets:
+#       api-host: ${{ secrets.REACT_APP_API_HOST_STAGE }}
+#       authentication-host: ${{ secrets.REACT_APP_AUTHENTICATION_HOST_STAGE }}
+#       domain: ${{ secrets.REACT_APP_DOMAIN_STAGE }}
```
Comment (pyphilia) [inline .github/workflows/deploy-stage.yml:69]: Delete?

File: .github/workflows/deploy-stage.yml [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+        # Set environment variables required to perform the build. These are only available to this step
+        env:
+          VITE_VERSION: ${{ github.event.client_payload.tag }}
+          VITE_GRAASP_DOMAIN: ${{ vars.VITE_GRAASP_DOMAIN }}
+          VITE_GRAASP_API_HOST: ${{ vars.VITE_GRAASP_API_HOST }}
+          VITE_GRAASP_AUTH_HOST: ${{ vars.VITE_GRAASP_AUTH_HOST }}
+          VITE_GRAASP_BUILDER_HOST: ${{ vars.VITE_GRAASP_BUILDER_HOST }}
+          VITE_SENTRY_ENV: ${{ vars.VITE_SENTRY_ENV }}
+          VITE_SENTRY_DSN: ${{ secrets.VITE_SENTRY_DSN }}
+          # VITE_GA_MEASUREMENT_ID: ${{ secrets.VITE_GA_MEASUREMENT_ID }}
```
Comment (pyphilia) [inline .github/workflows/deploy-stage.yml:38]: Delete?

File: cypress/plugins/index.ts [hunk 4]
```diff
@@ -1,6 +1,3 @@
-/* eslint-disable import/no-extraneous-dependencies */
-/* eslint-disable global-require */
-
 /// <reference types="cypress" />
 // ***********************************************************
 // This example plugins/index.js can be used to load plugins
```
Comment (pyphilia) [inline cypress/plugins/index.ts:3]: I think you can remove this file? I'm not sure

File: package.json [hunk 5]
```diff
@@ -44,42 +47,44 @@
     "papaparse": "5.4.1",
     "prop-types": "15.8.1",
     "qs": "6.11.2",
-    "react": "^17.0.2",
+    "react": "18.2.0",
```
Comment (pyphilia) [inline package.json:50]: 😮

File: package.json [hunk 6]
```diff
@@ -95,55 +100,49 @@
...
+    "eslint-plugin-import": "2.28.0",
     "eslint-plugin-jsx-a11y": "6.7.1",
-    "eslint-plugin-react": "7.32.2",
+    "eslint-plugin-react": "7.33.1",
+    "eslint-plugin-react-hooks": "4.6.0",
     "husky": "8.0.3",
-    "istanbul-lib-coverage": "3.2.0",
     "nyc": "15.1.0",
     "prettier": "3.0.1",
     "react-app-rewired": "2.2.1",
```
Comment (pyphilia) [inline package.json:139]: Do we still need that?

File: src/components/item/publish/CCLicenseDialog.jsx [hunk 7]
```diff
@@ -11,6 +9,8 @@ import {
 import { BUILDER } from '@graasp/translations';
 import { Button } from '@graasp/ui';
 
+import PropTypes from 'prop-types';
```
Comment (pyphilia) [inline src/components/item/publish/CCLicenseDialog.jsx:12]: Remove? Not a tsx file?

File: src/components/item/publish/CoEditorSettings.jsx [hunk 8]
```diff
@@ -1,13 +1,13 @@
...
 
 import { FormControlLabel, Radio, RadioGroup, Typography } from '@mui/material';
 
-import { useEffect, useState } from 'react';
-
 import { BUILDER } from '@graasp/translations';
 import { Loader } from '@graasp/ui';
 
+import { Record } from 'immutable';
+import PropTypes from 'prop-types';
```
Comment (pyphilia) [inline src/components/item/publish/CoEditorSettings.jsx:9]: Remove? Tsx file?

**Review by pyphilia — APPROVED** [review]
Incredible! I've added some comments 🚀

File: src/components/item/publish/CCLicenseDialog.jsx (hunk 7 above)
Comment (spaenleh) [reply src/components/item/publish/CCLicenseDialog.jsx:12]: It is actually a `jsx` file (vite requires this extension for files written in JS that contain JSX). But I will convert it to `tsx`.

File: package.json (hunk 5 above)
Comment (spaenleh) [reply package.json:50]: 🎆

File: cypress/plugins/index.ts (hunk 4 above)
Comment (spaenleh) [reply cypress/plugins/index.ts:4]: I am not sure, I will leave it just in case

**Pull request merged**